src/vellum/lib/utils/paginator.py

```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass
+from typing import Callable, Generic, TypeVar
+
+
+Result = TypeVar("Result")
+
+
+@dataclass
+class PaginatedResults(Generic[Result]):
+    count: int
+    results: list[Result]
+
+
+def get_all_results(
+    paginated_api: Callable[[int, int | None], PaginatedResults[Result]], page_size: int | None = None
+) -> list[Result]:
```

Comment:
As a consumer of this, I'd much prefer a return type of:
```suggestion
def get_all_results(
    paginated_api: Callable[[int, int | None], PaginatedResults[Result]], page_size: int | None = None
) -> Generator[Result, None, None]:
```

This way, I'm not _forced_ to exhaust the paginator.